src/view/editable-point-view.js

```diff
@@ -109,28 +115,147 @@ const createEditablePointTemplate = ({id = '', type = '', town = '', info = '',
       </section>
     </form>
   </li>`
-  );};
+);
+
+export default class EditablePointView extends SmartView {
+    #datepicker = []
 
-export default class EditablePointView extends AbstractView {
     #point = null
+    #availableOffers = []
+    #selectedOffers = []
 
     constructor(point) {
       super();
+      this._data = EditablePointView.parsePointToData(point);
       this.#point = point;
+      this.#availableOffers = getAvailableOffers(point.type);
+      this.#selectedOffers =  getSelectedOffers(point.offers, point.type);
+
+      this.#setInnerHandlers();
+      this.#setDatepicker();
     }
 
     get template() {
-      return createEditablePointTemplate(this.#point);
+      return createEditablePointTemplate(this._data);
+    }
+
+    static parsePointToData = (point) => {
+      const availableOffers = getAvailableOffers(point.type);
+      const selectedOffers = getSelectedOffers(point.offers, point.type);
+      return {...point, availableOffers, selectedOffers};
+    }
+
+    static parseDataTiPoint = (data) => {
```

Comment:
Наверное просто очепятка
```suggestion
    static parseDataToPoint = (data) => {
```